examples/finding_sunspots_using_stara.py

```diff
@@ -0,0 +1,94 @@
+"""
+============================
+Finding sunspots using STARA
+============================
+
+This example demonstrates the use of Sunspot Tracking And Recognition
+Algorithm (STARA) in detecting and plotting sunspots. More information
+on the algorithm can be found in [this](https://doi.org/10.1017/S1743921311014992) paper.
+If you wish to perform analysis over a large period of time we suggest to refer
+[this](https://gitlab.com/wtbarnes/aia-on-pleiades/-/blob/master/notebooks/tidy/finding_sunspots.ipynb)
+notebook implementation of the same algorithm using dask arrays.
+"""
+
+import astropy.units as u
+import matplotlib.pyplot as plt
+import sunpy.io._fits
+import sunpy.map
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+from sunkit_image.stara import stara
+
+###############################################################################
+# Firstly, let's download HMI continuum data from the Virtual Solar Observatory.
+
+query = Fido.search(a.Time("2023-01-01 00:00", "2023-01-01 00:01"), a.Instrument("HMI"), a.Physobs("intensity"))
+# We only download the first file in the search results in this example.
+file = Fido.fetch(query[0, 0])
+
+###############################################################################
+# Once the data is downloaded, we read the FITS file using`sunpy.map.Map`.
+
+hmi_map = sunpy.map.Map(file)
+
+###############################################################################
+# HMI maps are inverted, meaning that the solar north pole appears at the
+# bottom of the image. To correct this, we rotate each map in the MapSequence
+# using the ``rotate`` method with an order of 3. For detailed reason as to why, please refer this
+# `example <https://docs.sunpy.org/en/stable/generated/gallery/map_transformations/upside_down_hmi.html#sphx-glr-generated-gallery-map-transformations-upside-down-hmi-py>`__.
+
+cont_rotated = hmi_map.rotate(order=3)
+
+###############################################################################
+# Plot the rotated map.
+
+fig = plt.figure()
+ax = plt.subplot(projection=cont_rotated)
+im = cont_rotated.plot(axes=ax)
+
+
+###############################################################################
+# To reduce computational expense, we resample the continuum image to a lower
+# resolution. This step ensures that running the algorithm on the full-resolution
+# image is not overly computationally expensive.
+
+cont_rotated_resample = cont_rotated.resample((1024, 1024) * u.pixel)
```

Comment:
Yes this needs to be done if we wish to get the output quicker. Because the original image is ~4 times larger(4102*4102 pixels) than the resampled one. This causes a very big difference in runtime on a normal cpu ( ~15 mins verses 40 seconds)

Reply:
It takes 15 minutes to run strara?

Reply:
Yeah, it does take a lot of time on a cpu with 2.7 GHz

Reply:
What about on read the docs?

Reply:
The execution time for only the output from stara(on unsampled data) took 20 mins on the build_docs.

Reply:
Oh that is slow. Yeah keep the resample in here.

Can you add a comment about the performance difference?

Reply:
> Can you add a comment about the performance difference?

Sure, I'll add it.